tests/python/gaia-ui-tests/gaiatest/apps/base.py

```diff
@@ -148,7 +151,15 @@ def tap_element_from_system_app(self, element=None, add_statusbar_height=False):
     def keyboard(self):
         from gaiatest.apps.keyboard.app import Keyboard
         return Keyboard(self.marionette)
+    
+    @property
+    def name(self):
+        raise NotImplementedError("{} should implement the class attribute called 'name'".format(self.__class__)  )
 
+    @property
+    def manifest_url(self):
+        app_name_to_use = self.manifest_url_app if self.manifest_url_app else self.name.lower()
```

Comment:
How does manifest_url_app work in the case of dialer and contacts. They have the same manifestURL, right? But the manifest_url_app  of those are 
communications.{}/dialer/index.html'.format(self.DEFAULT_APP_HOSTNAME)
communications.{}/contacts/index.html'.format(self.DEFAULT_APP_HOSTNAME)
Afaik, the manifest_url property of those should result in app://communications.gaiamobile.org/manifest.webapp.
I don't see how manifest_url_app works here.